cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -613,9 +613,29 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             let to_bits = ty_bits(output_ty) as u8;
             let to_bits = std::cmp::max(32, to_bits);
             assert!(from_bits <= to_bits);
-            if from_bits < to_bits {
-                let signed = op == Opcode::Sextend;
-                let rd = get_output_reg(ctx, outputs[0]).only_reg().unwrap();
+
+            let signed = op == Opcode::Sextend;
+            let dst = get_output_reg(ctx, outputs[0]);
+            let src =
+                if let Some(extract_insn) = maybe_input_insn(ctx, inputs[0], Opcode::Extractlane) {
+                    put_input_in_regs(
+                        ctx,
+                        InsnInput {
+                            insn: extract_insn,
+                            input: 0,
+                        },
+                    )
+                } else {
+                    put_input_in_regs(ctx, inputs[0])
+                };
+
+            let needs_extend = from_bits < to_bits && to_bits <= 64;
+            // For i128, we want to extend the lower half, except if it is already 64 bits.
+            let needs_lower_extend = to_bits > 64 && from_bits < 64;
```

Comment:
Can we clarify the logic a bit by adding a `let pass_through_lower = to_bits > 64 && !needs_lower_extend` here, then moving the move-emission below (will comment separately)?